controllers/noobaa-rules.yaml

```
@@ -1,104 +1,96 @@
        - expr: |
-           NooBaa_total_usage{namespace="redhat-data-federation"}
+           max without(instance) (NooBaa_num_unhealthy_namespace_resources{namespace="redhat-data-federation"})
          labels:
            origin: data-federation-service
-         record: noobaa_total_usage
+         record: namespace:noobaa_unhealthy_namespace_resources:max
        - expr: |
-            NooBaa_num_unhealthy_namespace_resources{namespace="redhat-data-federation"}
+           max without(instance) (NooBaa_num_namespace_resources{namespace="redhat-data-federation"})
          labels:
            origin: data-federation-service
-         record: noobaa_num_unhealthy_namespace_resources
+         record: namespace:noobaa_namespace_resources:max
        - expr: |
-           NooBaa_num_namespace_resources{namespace="redhat-data-federation"}
+           max without(instance) (NooBaa_num_namespace_buckets{namespace="redhat-data-federation"})
          labels:
            origin: data-federation-service
-         record: noobaa_num_namespace_resources
+         record: namespace:noobaa_namespace_buckets:max
        - expr: |
-           NooBaa_namespace_resource_status{namespace="redhat-data-federation"}
+           count(NooBaa_namespace_bucket_status{namespace="redhat-data-federation"} == 0)
          labels:
            origin: data-federation-service
-         record: noobaa_namespace_resource_status
-       - expr: |
-           NooBaa_namespace_bucket_status{namespace="redhat-data-federation"}
-         labels:
-           origin: data-federation-service
-         record: noobaa_namespace_bucket_status
+         record: namespace:unhealthy_noobaa_namespace_buckets:count
    - name: noobaa-internal.rules
      rules:
        - expr: |
-           NooBaa_odf_health_status{namespace="redhat-data-federation"}
+           max without (instance) (NooBaa_odf_health_status{namespace="redhat-data-federation"})
          labels:
-           system_type: NooBaa
-           system_vendor: Red Hat
            origin: data-federation-service
-         record: noobaa_system_health_status
+         record: namespace:noobaa_system_health_status:max
    - name: bucket-state-alert.rules
      rules:
-       - alert: BucketPolicyErrorState
+       - alert: NooBaaBucketPolicyErrorState
          annotations:
            description: A Bucket Policy {{ $labels.bucket_name }} is in error
              state for more than 5m
-           message: A Bucket Policy Is In Error State
-           severity_level: warning
-           storage_type: NooBaa
+           summary: A Bucket Policy Is In Error State
+         # alert if this metric was up for less than 80% in the last 5 minutes, on average.
+         # refer https://github.com/red-hat-storage/mcg-osd-deployer/pull/67#discussion_r930856351 for more details.
          expr: |
-           NooBaa_namespace_bucket_status{bucket_name=~".*"} == 0
+           avg_over_time(NooBaa_namespace_bucket_status[5m]) < 0.8
```

(nit) if you want to only look at the last sample, last_over_time is even better.

```suggestion
           last_over_time(NooBaa_namespace_bucket_status[5m]) < 1
```

---

I think we want to look over the range instead of just the last value, right @cloudbehl?

---

last_over_time() will return the last known value in the range. E.g. `[0, 0, 0, 0, 1]` will return 1 where avg_over_time() would return 0.2.

---

I agree, last_over_time will make sense.